k8s/deployment.yaml

```diff
@@ -0,0 +1,54 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: palworld-server
+  name: palworld-server
+  namespace: default
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: palworld-server
+  template:
+    metadata:
+      labels:
+        app: palworld-server
+    spec:
+      containers:
+      - env:
```

Comment:
Maybe do environment variables like this:
```
        - name: ADMIN_PASSWORD
          valueFrom:
            secretKeyRef:
              name: palworld-secrets
              key: rconPassword
        envFrom:
        - configMapRef:
            name: palworld-cm
```

secrets.palworld.yml:
```
apiVersion: v1
kind: Secret
metadata:
  name: palworld-secrets
type: Opaque
stringData:
  rconPassword: yourRconPassword
```

configmap-palworld.yml
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: palworld-cm
data:
  PUID: "1000"
  PGID: "1000"
  PORT: "8211" # Optional but recommended
  PLAYERS: "16" # Optional but recommended
  MULTITHREADING: "false"
  RCON_ENABLED: "true"
  RCON_PORT: "25575"
  COMMUNITY: "false"  # Enable this if you want your server to show up in the community servers tab, USE WITH SERVER_PASSWORD!
  # Enable the environment variables below if you have COMMUNITY=true
  # SERVER_PASSWORD: "yourServerPassword"
  SERVER_NAME: "Server_Name" # Note, it seems spaces are not supported, so use underlines.
```